pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1001,3 +1013,94 @@ func (b *SlirpPodInterface) setCachedVIF(pid, name string) error {
 func (s *SlirpPodInterface) setCachedInterface(pid, name string) error {
 	return nil
 }
+
+type MacvtapPodInterface struct {
+	vmi              *v1.VirtualMachineInstance
+	vif              *VIF
+	iface            *v1.Interface
+	virtIface        *api.Interface
+	domain           *api.Domain
+	podInterfaceName string
+}
+
+func (m *MacvtapPodInterface) discoverPodNetworkInterface() error {
+	if len(m.vif.MAC) == 0 {
+		// Get interface MAC address
+		mac, err := Handler.GetMacDetails(m.podInterfaceName)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to get MAC for %s", m.podInterfaceName)
+			return err
+		}
+		m.vif.MAC = mac
+	}
+
+	return nil
+}
+
+func (m *MacvtapPodInterface) preparePodNetworkInterfaces() error {
+	m.virtIface.MAC = &api.MAC{MAC: m.vif.MAC.String()}
+	m.virtIface.Target = &api.InterfaceTarget{
+		Device:  m.podInterfaceName,
+		Managed: "no",
+	}
+	return nil
+}
+
+func (m *MacvtapPodInterface) decorateConfig() error {
+	ifaces := m.domain.Spec.Devices.Interfaces
+	for i, iface := range ifaces {
+		if iface.Alias.Name == m.iface.Name {
+			ifaces[i].MTU = m.virtIface.MTU
```

Comment:
Couldn't find any place the set `virtIface.MTU` for macvtap. So this code seems to be redundant.